action.yml

```diff
@@ -11,9 +15,12 @@ runs:
       with:
         python-version: '3.10'
 
-    - run: python3 -m merge_checks
+    - run: pip install .
```

Comment:
So, I don't quite understand why this stuff in necessary. It seems very complicated to me. I see from the description that you faced some `PYTHONPATH` problems. But why the following (normal?) structure didn't work for you?


```
python3 merge_checks_runner.py

(no __init__.py at the root of the action)

merge_checks_runner.py ( doesn't do anything other than import merge_checks.main ; main() )

merge_checks/commit_checks.py
merge_checks/commit_status_setter.py
merge_checks/...
merge_checks/tests/test_blah.py 
merge_checks/tests/test_....py
```

Reply:
So I have tried it myself between the meetings in a different branch, and now I understand better what problems you were trying to solve, but I think that there are other problems with your solution and conversely the problem is not really a big problem, and can be solved differently.


https://github.com/moneymeets/action-merge-checks/compare/feature/MD-5393-separate-workflow-from-merge-check-status-test

I would suggest keep using the package and pack the runner in `__main__` and do few other things differently. But I think it will take a lot of time to write down everything in detail and unambiguously communicate about it.

My suggestion would be to have a meeting to hopefully quickly resolve misunderstandings and discuss the solution. I've sent you a proposal per outlook. Hopefully this would work for you.

Reply:
I'm happy to find a good solution for this together. I just want to make sure that we have a common understanding of the problems we need to solve:

- The merge checks module, as defined here need, needs to be executed in the checked out code of the repository that called the action 
- It should be maximally unlikely (or better impossible) that instead of the merge checks module, a module of the repository that called the action is run instead
- The PyGitHub dependency needs to be installed and available when running the merge checks module

I have described the first two points in more detail in the description of this PR, but I'm concerned that the PyGitHub dependency is being overlooked. Your current suggestion only works since you are running the merge checks from this very repository. It will fail in all others.

Reply:
@zyv and I had a closer look at this together just now. After trying a couple of different things, we were able to come up with a good solution of how to completely avoid possible path conflicts. 

The concept is as follows:

- Dependencies of the merge checks get install globally by pip as before.
- We have added a line to pyproject.toml to instruct the installation of merge_checks_runner.run as a script with name `merge_checks_runner`.
- To run the merge checks, we simply call `merge_checks_runner` in the working directory of repository that triggered the workflow.

We have also discussed using poetry to setup and manage the dependencies of the merge checks. This would require changes to our poetry setup action and would increase the merge check run time. In the end, we agreed that since we are only running the merge checks (and not any code of the repository that triggered the action), the pip installation of dependencies in this isolated workflow run is acceptable.

To merge this pull request, we agreed to use the method to temporarily disable the required checks for this repository. After this, for subsequent changes, merge checks should be properly reported also in this repository, as we are now avoiding any path conflicts.

I'm leaving this open for a bit for visibility, but the points discussed in this thread are resolved. @marns93 @v-the-cmd, do you have any questions on this? 

﻿

Reply:
I would prefer to avoid installing stuff in the target environment, but that solution looks more-or-less okay to me, so no objections.

Reply:
Yeah, we discussed that, but the job does the checks and nothing else, and will be disposed afterwards, so we agreed that it's better than doing the complicated venv dance or giving up upon dependencies or some such...

Reply:
@felix11h Thank you for the explanation. I have no more questions. The solution is okay for me.